src/NuGet.Clients/NuGet.CommandLine/Commands/PushCommand.cs

```diff
@@ -74,6 +74,15 @@ private string ResolveSource(string packagePath, string configurationDefaultPush
         {
             var source = Source;
 
+            if (string.IsNullOrEmpty(source))
+            {
+                source = Settings.GetValue(ConfigurationConstants.Config, ConfigurationConstants.DefaultPushSource);
+            }
+
+            if (string.IsNullOrEmpty(source))
```

Comment:
Why is configurationDefaultPushSource not reading the setting, and we duplicate the reading here?